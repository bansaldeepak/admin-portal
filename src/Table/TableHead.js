import React from 'react';
import PropTypes from 'prop-types';

import { withStyles } from '@material-ui/core/styles';
import TableCell from '@material-ui/core/TableCell';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import TableSortLabel from '@material-ui/core/TableSortLabel';
import Checkbox from '@material-ui/core/Checkbox';
import Tooltip from '@material-ui/core/Tooltip';

const tableHeadStyles = () => ({
  head: {
    background: '#f5f5f5'
  },
  tableCell: {
    padding: '8px'
  }
});

const EnhancedTableHead = (props) => {
  const { columns, classes, numSelected, rowCount, onSelectAllClick, selectable, orderBy, order } = props;

  const createSortHandler = property => event => {
    props.onRequestSort(event, property);
  };

  const headerRequired = columns.find(column => {
    return column.label;
  });

  return (headerRequired) ? (
    <TableHead className={classes.head}>
      <TableRow>
        {columns.map(column => {
          return (
            (selectable && column.type === 'select') ? (
              <TableCell
                padding='checkbox'
                classes={{
                  root: classes.tableCell
                }}
              >
                <Checkbox
                  indeterminate={numSelected > 0 && numSelected < rowCount}
                  checked={numSelected === rowCount}
                  onChange={onSelectAllClick}
                />
              </TableCell>
            ) : (
              (column.show === undefined || column.show) ? (
                <TableCell
                  key={`${(new Date()).getTime()}-table-head-${column.name}`}
                  padding={column.disablePadding ? 'none' : 'default'}
                  sortDirection={orderBy === column.name ? order : false}
                  classes={{
                    root: classes.tableCell
                  }}
                >
                  {
                    (column.sort) ? (
                      <Tooltip
                        title='Sort'
                        placement={column.numeric ? 'bottom-end' : 'bottom-start'}
                        enterDelay={300}
                      >
                        <TableSortLabel
                          active={orderBy === column.name}
                          direction={order}
                          onClick={createSortHandler(column.name)}
                        >
                          {column.label}
                        </TableSortLabel>
                      </Tooltip>
                    ) : column.label
                  }
                </TableCell>
              ) : null
            )
          );
        })}
      </TableRow>
    </TableHead>
  ) : null;
}

EnhancedTableHead.defaultProps = {
  order: 'desc',
  rowCount: 0,
  selectable: false,
  orderBy: 'id',
  numSelected: 0,
  columns: []
};

EnhancedTableHead.propTypes = {
  numSelected: PropTypes.number.isRequired,
  onRequestSort: PropTypes.func.isRequired,
  onSelectAllClick: PropTypes.func.isRequired,
  order: PropTypes.string.isRequired,
  // orderBy: PropTypes.string.isRequired,
  rowCount: PropTypes.number.isRequired,
  classes: PropTypes.object.isRequired
};

export default withStyles(tableHeadStyles)(EnhancedTableHead);