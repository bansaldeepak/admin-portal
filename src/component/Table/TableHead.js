import React from "react";
import PropTypes from 'prop-types';

import { withStyles } from '@material-ui/core/styles';
import TableCell from '@material-ui/core/TableCell';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import TableSortLabel from '@material-ui/core/TableSortLabel';
import Checkbox from '@material-ui/core/Checkbox';
import Tooltip from '@material-ui/core/Tooltip';

const tableHeadStyles = theme => ({
  head: {
    background: '#f5f5f5',
  },
  tableCell: {
    padding: '4px 8px 4px 8px'
  }
});

const tableHeadOptions = {
  order: "asc",
  orderBy: "id",
  rowCount: 0,
  numSelected: 0,
  columns: []
}

class EnhancedTableHead extends React.Component {
  constructor(props) {
    super(props);

    this.state = Object.assign(tableHeadOptions, props);
  }

  createSortHandler = property => event => {
    this.props.onRequestSort(event, property);
  };

  render = () => {
      const {columns, classes} = this.props;
      
      return (
        <TableHead className={classes.head}>
          <TableRow>
            {columns.map(column => {
              return (
                (column.name === "selectable") ?
                (<TableCell 
                  padding="checkbox"
                  classes={{
                    root: classes.tableCell
                  }}
                ><Checkbox indeterminate={this.props.numSelected > 0 && this.props.numSelected < this.props.rowCount} checked={this.props.numSelected === this.props.rowCount} onChange={this.props.onSelectAllClick}/></TableCell>) :
                (
                  (column.show === undefined || column.show) ? (
                    <TableCell
                      key={`${(new Date()).getTime()}-table-head-${column.name}`}
                      padding={column.disablePadding ? 'none' : 'default'}
                      sortDirection={this.props.orderBy === column.name ? this.props.order : false}
                      classes={{
                        root: classes.tableCell
                      }}
                    >
                      {(column.sort) ? (
                          <Tooltip
                            title="Sort"
                            placement={column.numeric ? 'bottom-end' : 'bottom-start'}
                            enterDelay={300}
                          >
                            <TableSortLabel
                                active={this.props.orderBy === column.name}
                                direction={this.props.order}
                                onClick={this.createSortHandler(column.name)}
                            >
                                {column.label}
                            </TableSortLabel>
                          </Tooltip>
                        ) : column.label
                      }
                    </TableCell>
                  ) : null
                )
              );
            }, this)}
          </TableRow>
        </TableHead>
      );
  }
}

EnhancedTableHead.defaultProps = {
  order: 'desc',
  rowCount: 0
};

EnhancedTableHead.propTypes = {
  numSelected: PropTypes.number.isRequired,
  onRequestSort: PropTypes.func.isRequired,
  onSelectAllClick: PropTypes.func.isRequired,
  order: PropTypes.string.isRequired,
  // orderBy: PropTypes.string.isRequired,
  rowCount: PropTypes.number.isRequired,
  classes: PropTypes.object.isRequired,
};

export default withStyles(tableHeadStyles)(EnhancedTableHead);