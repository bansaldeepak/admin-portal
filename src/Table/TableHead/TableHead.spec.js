import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { shallowToJson } from "enzyme-to-json";
import TableHead from "./TableHead";

configure({ adapter: new Adapter() });

describe("TableHead", () => {
  it("Render TableHead with default options", () => {
    const component = shallow(<TableHead />);
    const tree = shallowToJson(component);
    expect(tree).toMatchSnapshot();
  });
});
