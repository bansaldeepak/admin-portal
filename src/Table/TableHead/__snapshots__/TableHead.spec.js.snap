// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHead Render TableHead with default options 1`] = `
<EnhancedTableHead
  classes={
    Object {
      "head": "EnhancedTableHead-head-1",
      "tableCell": "EnhancedTableHead-tableCell-2",
    }
  }
  columns={Array []}
  numSelected={0}
  order="desc"
  orderBy="id"
  rowCount={0}
  selectable={false}
/>
`;
